### PR TITLE
Switch route URLs

### LINKED ISSUE #142: Octoclairvoyant base domain is the homepage
## What feature would you like to have?

Make the https://octoclairvoyant.vercel.app/ base domain our landing page.

Use something like `https://octoclairvoyant.vercel.app/comparator` to show the repository compare view.
Once a user has selected a repository, we can use the `comparator/?query-string` pattern.

Proposed URL when a repository has been selected:
`https://octoclairvoyant.vercel.app/comparator/?repo=renovatebot%2Frenovate&from=25.38.0&to=latest`

## What problem will be solved by having the feature?

In my wireframes I've removed the about page, on the assumption that we can use https://octoclairvoyant.vercel.app/ as the landing page.

In the current implementation a user will go straight to the repository comparator select screen, without knowing what the site is about. If they land on the homepage first, they can learn what Octoclairvoyant does and the features it has.

## Do you want to help making the feature?

I don't know how to make this change.

~This feature will be a BREAKING CHANGE for the comparator query string URLs that users have shared/bookmarked.~
No it wont, we will redirect end-users, so old saved links will continue working.

### PR DESCRIPTION
**Pull request by Belco90**
## Changes

- Move `/` page to `/comparator` page
- Move `/about` page to `/` page
- Update existing links to reference new URLs
- Remove navigation links from the header
- Add a link to the main text in the header
- Redirect previous comparator URLs to the new one

## Context

Closes #142

### CODE REVIEW

File: next.config.js
Comment (Belco90) [inline next.config.js:18]: This is the definition to redirect from previous `/` to `/comparator` only when necessary (i.e. when the URL has a querystring `repo`). By this config:

- accessing `/` will stay in `/` URL
- accessing `/?repo=renovatebot%2Frenovate&from=25.54.0&to=latest` will redirect to `/comparator?repo=renovatebot%2Frenovate&from=25.54.0&to=latest` URL
- accessing `/?repo=renovatebot%2Frenovate` will redirect to `/comparator?repo=renovatebot%2Frenovate` URL
- accessing `/?foo=bar` will stay in `/?foo=bar`

File: tsconfig.json
Comment (Belco90) [inline tsconfig.json:28]: This is an additional change since I saw in Next.js docs this is the preferred way to declare additional types.

**Review by Belco90 — COMMENTED** [review]
Probably easier to review by commits.

File: next.config.js
Comment (HonkingGoose) [reply next.config.js:18]: I've tested this and it seems to work well! 😄

File: src/pages/index.tsx
Comment (HonkingGoose) [inline src/pages/index.tsx:76]: I think using the term Homepage is clearer that About. What do you think?

File: src/components/Header.tsx
Comment (HonkingGoose) [inline src/components/Header.tsx:53]: I've added some basic `_hover`, `_focus` and `_active` styles.

I think this looks a bit more "on brand" than just a black underline. But maybe you liked the simplicity of the black line, so feel free to revert, or suggest changes. 😉

Comment (Belco90) [reply src/components/Header.tsx:53]: Much better now, thanks! 😄 

I'd like to mention if we can avoid other people than the PR creator pushing changes? It's a little bit intrusive in the sense of I'd have to revert those changes if we are not happy with it, or introduces errors, etc. Also, it can even prevent the PR from being merged _officially_ since some VCS prevent PRs from being approved by people who pushed changes to the branch (we wouldn't be blocked since I have admin permissions, but still).

I think it's better to add a comment with proper suggestions to be pushed by the creator of the PR. What do you think?

File: src/pages/index.tsx
Comment (Belco90) [reply src/pages/index.tsx:76]: Yes, you are right! I'll rename it later.

**Comment by HonkingGoose** [comment]
> Much better now, thanks! 😄
> 
> I'd like to mention if we can avoid other people than the PR creator pushing changes? It's a little bit intrusive in the sense of I'd have to revert those changes if we are not happy with it, or introduces errors, etc. Also, it can even prevent the PR from being merged _officially_ since some VCS prevent PRs from being approved by people who pushed changes to the branch (we wouldn't be blocked since I have admin permissions, but still).
> 
> I think it's better to add a comment with proper suggestions to be pushed by the creator of the PR. What do you think?

You're right, pushing commits to your branch was a silly idea of me! I did not consider that doing this might break the ability to merge the PR with the proper review approval. And it's just confusing if new commits suddenly appear on your branch...

I apologize for messing with your branch.

I'll add a comment with suggestions in future PRs, so that you can implement them yourself. 😉

**Review by HonkingGoose — APPROVED** [review]
Let's see if my approval will work now that I've pushed commits to this branch...

---

EDIT: I'll let you decide if you want to merge this as-is, or if you want to revert my commits, and have me make a new PR with the changes. 😉

**Comment by Belco90** [comment]
> Let's see if my approval will work now that I've pushed commits to this branch...
> 
> 
> 
> ---
> 
> 
> 
> EDIT: I'll let you decide if you want to merge this as-is, or if you want to revert my commits, and have me make a new PR with the changes. 😉 

It should work tho! We were in the same situation before and GitHub allowed us to approve it and merge with no problem.

Merging now! It's not worth to extract anything into another branch.

Thanks!

**Comment by Belco90** [comment]
Just to confirm: it seems GitHub doesn't have an option to disable approvals from reviewers who pushed to the PR branch, so this seems a feature only available on GitLab and BitBucket.

File: src/pages/index.tsx
Comment (Belco90) [reply src/pages/index.tsx:76]: I forgot about this! You flagged it as resolved so I didn't realize this was pending. Let me fix it in a separate PR.

Comment (Belco90) [reply src/pages/index.tsx:76]: Ah damn, it was renamed already 🤦 . I thought it was pending.

Comment (Belco90) [reply src/pages/index.tsx:76]: I have renamed the component for comparator page tho https://github.com/Belco90/octoclairvoyant/commit/d0482b61910e3e615215acf181823891ea5ead3b

**Pull request merged**
